toolchain/check/impl_lookup.cpp

```diff
@@ -1036,8 +1032,8 @@ auto EvalLookupSingleImplWitness(Context& context, SemIR::LocId loc_id,
       // new impl decl was written that changes the result.
       if (mode != EvalImplLookupMode::RecheckPoisonedLookup &&
           result.has_final_value() &&
-          !IsImplEffectivelyFinal(context,
-                                  context.impls().Get(candidate.impl_id))) {
+          context.impls().Get(candidate.impl_id).final_kind ==
```

Comment:
Can we pass the `Impl&` to `GetWitnessIdForImpl` and avoid a second lookup here?

Reply:
Done.